Source/WebCore/rendering/RenderFlexibleBox.cpp

```diff
@@ -2202,16 +2206,15 @@ void RenderFlexibleBox::adjustAlignmentForChild(RenderBox& child, LayoutUnit del
     
 void RenderFlexibleBox::alignChildren(const Vector<LineContext>& lineContexts)
 {
-    // Keep track of the space between the baseline edge and the after edge of
-    // the box for each line.
-    Vector<LayoutUnit> minMarginAfterBaselines;
-    
+    bool containerHasWrapReverse = style().flexWrap() == FlexWrap::Reverse;
+    auto shouldAdjustItemTowardsCrossAxis = [this, containerHasWrapReverse](const RenderBox& box, ItemPosition alignment) { 
+        return (containerHasWrapReverse && alignment == ItemPosition::Baseline && !hasAutoMarginsInCrossAxis(box)) || (alignment == ItemPosition::LastBaseline && !containerHasWrapReverse); 
```

Comment:
Why check `!hasAutoMarginsInCrossAxis(box)` for the 1st baseline but not for the last one?

Reply:
If it should be checked in both cases, then the logic could be shared with another conditional:

```cpp
    auto something = [containerHasWrapReverse](ItemPosition alignment) { 
        return (containerHasWrapReverse && alignment == ItemPosition::Baseline) || (alignment == ItemPosition::LastBaseline && !containerHasWrapReverse); 
    };
// ...
            if (something(position))
                minMarginAfterBaseline = std::min(minMarginAfterBaseline, availableAlignmentSpaceForChild(lineCrossAxisExtent, flexItem.box) - offset);
// ...
                if (!hasAutoMarginsInCrossAxis(flexItem.box) && something(alignmentForChild(flexItem.box)))
                    adjustAlignmentForChild(flexItem.box, minMarginAfterBaseline);
```

Reply:
Yes it should be for both. I made the adjustment, but let me know if you see any other issues with this

Reply:
Removed now